services/web/server/src/simcore_service_webserver/projects/projects_nodes_handlers.py

```diff
@@ -263,6 +267,17 @@ async def stop_node(request: web.Request) -> web.Response:
     """Has only effect on nodes associated to dynamic services"""
     req_ctx = RequestContext.parse_obj(request)
     path_params = parse_request_path_parameters_as(_NodePathParams, request)
+    project_db_api: ProjectDBAPI = request.app[APP_PROJECT_DBAPI]
+
+    save_state = await project_db_api.has_permission(
+        user_id=req_ctx.user_id, project_uuid=path_params.project_id, permission="write"
+    )
+    try:
+        user_role = await get_user_role(request.app, req_ctx.user_id)
+    except UserNotFoundError:
```

Comment:
so, ``has_permission`` can also raise ``UserNotFoundError`` and ``ProjectNotFound`` (see ``__load_user_groups`` inside ``has_permissions``)

NOTE: If the user comes from the request (i.e. ``req_ctx.user_id``) I guess there are some guarantees that it exists since it is put there by the login middleware (who checks for auth, etc)

Reply:
Thanks for the tip. Will remove check here.